dataland-frontend/src/components/general/apiKey/ApiKeyCard.vue

```diff
@@ -68,17 +64,26 @@ export default defineComponent({
   }),
   computed: {
     expiryDateInDays() {
-      return calculateDaysFromNow(this.expiryDate);
+      return calculateDaysFromNow(this.expiryDate * 1000);
```

Comment:
1) I delete the variable const d since is an unnecessary value rewrite.
2) startDate is optional. You can enter it default value is  "today"
3) Ok. It will be "endDateInMilliseconds" 
4) I use function "calculateDaysFromNow" to set expireTimeDays. It's useful

Reply:
1. Ok 👍 
2. Yes but as I've checked the code I found no place where you actually pass a startDate to the function. So you might as well drop the optional startDate param for now.
3. Ok sounds good 👍 
4. Ok agree. I overlooked that you actually use it in "CreateApiKeyCard.vue" to get the dayValid for your emit.

Reply:
ok

Reply:
about point 2:

You should not do ```const today = new Date().getTime();``` at the top of the ```DateFormatUtils.ts``` file.

Reason: If you do it like that, then the "today" time is just calculated once when a method from the ```DateFormatUtils.ts``` is used for the first time.
This is not good because a user might not close his browser for some days and then has the wrong "today" date.

I suggest calculating the "today" time on each call of ```calculateDaysFromNow()``` inside of the function. so it is guaranteed that it's always the current/correct "today" time.

Reply:
Furthermore I suggest to call it something like ```currentUtcDateInMilliseconds``` to make clear that we use the same time fromat (UTC) as the expiryDate in the backend uses, to avoid problems with time zones.

Reply:
Will his session expire after some time?

Reply:
Might be that the fact that his login expires and his re-login triggers the .ts-files to be pulled from scratch makes the today date to be recalculated.
But I don't have a good feeling in trusting in that if we have the possibility to avoid this problem, because I don't have a deep understanding of how and when Vite loads our .ts-files.

Reply:
done

Reply:
Nice.   Is it ok for you to rename from ```today``` to ``` currentUtcDateInMilliseconds``` or smiliar to show the reader of the code directly that this is a Unix time in Milliseconds?

Reply:
changed